libraries/standard/http/test/test-HTTPClient_AddHeader.c

```diff
@@ -1,9 +1,182 @@
+#include <string.h>
+
 #include "common.h"
 
 /* Functions are pulled out into their own C files to be tested as a unit. */
+#include "_addHeader.c"
 #include "HTTPClient_AddHeader.c"
 
+/* Template HTTP header fields and values. */
+#define HTTP_TEST_HEADER_FIELD               "Authorization"
+#define HTTP_TEST_HEADER_FIELD_LEN           ( sizeof( HTTP_TEST_HEADER_FIELD ) - 1 )
+#define HTTP_TEST_HEADER_VALUE               "None"
+#define HTTP_TEST_HEADER_VALUE_LEN           ( sizeof( HTTP_TEST_HEADER_VALUE ) - 1 )
+/* Template for first line of HTTP header. */
+#define HTTP_TEST_HEADER_REQUEST_LINE        "GET / HTTP/1.1 \r\n"
+#define HTTP_TEST_HEADER_REQUEST_LINE_LEN    ( sizeof( HTTP_TEST_HEADER_REQUEST_LINE ) - 1 )
+#define HTTP_REQUEST_HEADERS_INITIALIZER     { 0 }
+/* Default size for request buffer. */
+#define HTTP_TEST_BUFFER_SIZE                ( 100 )
+
+/* Length of the following template HTTP header.
+ *   <HTTP_TEST_HEADER_REQUEST_LINE> \r\n
+ *   <HTTP_TEST_HEADER_FIELD>: <HTTP_TEST_HEADER_VALUE> \r\n
+ *   \r\n
+ * This is used to initialize the correctHeader string. */
+#define HTTP_TEST_SUFFICIENT_HEADER_LEN   \
+    ( HTTP_TEST_HEADER_REQUEST_LINE_LEN + \
+      HTTP_TEST_HEADER_FIELD_LEN +        \
+      HTTP_HEADER_FIELD_SEPARATOR_LEN +   \
+      HTTP_TEST_HEADER_VALUE_LEN +        \
+      HTTP_HEADER_LINE_SEPARATOR_LEN +    \
+      HTTP_HEADER_LINE_SEPARATOR_LEN )
+
 int main()
 {
-    return 0;
-}
\ No newline at end of file
+    HTTPRequestHeaders_t reqHeaders = HTTP_REQUEST_HEADERS_INITIALIZER;
+    HTTPRequestHeaders_t reqHeadersDflt = HTTP_REQUEST_HEADERS_INITIALIZER;
+    HTTPStatus_t test_err = HTTP_NOT_SUPPORTED;
+    uint8_t buffer[ HTTP_TEST_BUFFER_SIZE ] = { 0 };
+    uint8_t smallBuffer[ HTTP_TEST_SUFFICIENT_HEADER_LEN - 1 ] = { 0 };
+    char correctHeader[ HTTP_TEST_BUFFER_SIZE ] = { 0 };
+    size_t correctHeaderLen = HTTP_TEST_SUFFICIENT_HEADER_LEN;
+    struct Header
+    {
+        char field[ 100 ];
+        size_t fieldLen;
+        char value[ 100 ];
+        size_t valueLen;
+    }
+    header;
+
+/* Write template header field and value to a struct to pass as
+ * parameters to HTTPClient_AddHeader() method. */
+#define fillHeaderStructTemplate()                                    \
+    do {                                                              \
+        memcpy( header.field,                                         \
+                HTTP_TEST_HEADER_FIELD, HTTP_TEST_HEADER_FIELD_LEN ); \
+        header.fieldLen = HTTP_TEST_HEADER_FIELD_LEN;                 \
+        memcpy( header.value,                                         \
+                HTTP_TEST_HEADER_VALUE, HTTP_TEST_HEADER_VALUE_LEN ); \
+        header.valueLen = HTTP_TEST_HEADER_VALUE_LEN;                 \
+    }                                                                 \
+    while( 0 )
+
+#define reset()                                                      \
```

Comment:
These macros can be defined outside `main()`.
The `main()` should only contain test code

Reply:
that is not how it is defined in mock4thewin's example